src/Sulu/Bundle/MediaBundle/Entity/CollectionRepository.php

```diff
@@ -233,4 +234,24 @@ public function findCollectionBreadcrumbById($id)
             return [];
         }
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function findCollectionByKey($key)
+    {
+        $queryBuilder = $this->createQueryBuilder('collection')
+            ->leftJoin('collection.meta', 'collectionMeta')
+            ->leftJoin('collection.defaultMeta', 'defaultMeta')
+            ->where('collection.key = :key');
+
+        $query = $queryBuilder->getQuery();
+        $query->setParameter('key', $key);
+
+        try {
+            return $query->getSingleResult();
+        } catch (UnexpectedResultException $ex) {
```

Comment:
the `UnexpectedResultException` is the base class of `NonUniqueResultException` and `NoResultException` so it would catch both. should i change it anyway?